lib/input/reader/amqp_1.go

```diff
@@ -193,14 +246,137 @@ func (a *AMQP1) ReadWithContext(ctx context.Context) (types.Message, AsyncAckFn,
 
 	msg.Append(part)
 
+	var done chan struct{}
+	if a.conf.RenewLock {
+		done = a.startRenewJob(amqpMsg)
+	}
+
 	return msg, func(ctx context.Context, res types.Response) error {
+		if done != nil {
+			close(done)
```

Comment:
It might be a bit paranoid but there's no explicit guarantee that the ack func is only called once. However, it's probably overkill to use something like `sync.Once` here so I'd say setting `done = nil` after the close is enough.